### PR TITLE
Add IS_IMPLICIT_ACCOUNT primitive for Seoul protocol v023

### PR DESCRIPTION
**Pull request by Frank-en-stein**
### Summary
Adds support for the `IS_IMPLICIT_ACCOUNT` opcode (0x9E) in the Seoul protocol, fixing the tzwatch indexer halt on Ghostnet caused by unknown primitive errors. Includes comprehensive unit tests and maintains backward compatibility.

### Changes
- Add `I_IS_IMPLICIT_ACCOUNT` opcode (0x9E) to opcodes
- Update `IsValid()` function to include new primitive
- Add unit tests with positive and negative cases
- Fix gofmt formatting issues

### References
Closes: https://github.com/trilitech/tzgo/issues/57  
Closes: https://linear.app/tezos/issue/L1INDX-46/gh-issue-57-add-is-implicit-account-primitive-for-seoul-protocol-v0230

### CODE REVIEW

File: micheline/opcodes_test.go [hunk 1]
```diff
@@ -0,0 +1,376 @@
+// Copyright (c) 2020-2024 Blockwatch Data Inc.
+// Author: alex@blockwatch.cc
```
Comment (huancheng-trili) [inline micheline/opcodes_test.go]: Please update the copyright lines. Same for the other new file.
Like this: https://github.com/trilitech/tzgo/blob/4bda970bf251dafadb40065a45767168ea26fdec/rpc/attestations_aggregate_test.go#L1
Though the company name should probably be `TriliTech Ltd.`

File: micheline/opcodes_test.go [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+			t.Errorf("ParseOpCode(IS_IMPLICIT_ACCOUNT) error: %v", err)
+		}
+		if parsed != I_IS_IMPLICIT_ACCOUNT {
+			t.Errorf("ParseOpCode(IS_IMPLICIT_ACCOUNT) = %v, expected %v",
+				parsed, I_IS_IMPLICIT_ACCOUNT)
+		}
+	})
+}
+
+func TestOpCodeSequence(t *testing.T) {
```
Comment (huancheng-trili) [inline micheline/opcodes_test.go]: Why do we need this? If you want to make sure that existing codes are never mutated, I think it's better to create a fixture file or so with a full code-name mapping.

Comment (Frank-en-stein) [reply micheline/opcodes_test.go]: Better removing it
